sklearn/manifold/tests/test_isomap.py

```diff
@@ -14,66 +14,98 @@
 eigen_solvers = ['auto', 'dense', 'arpack']
 path_methods = ['auto', 'FW', 'D']
 
+N_per_side = 5
+Npts = N_per_side ** 2
 
-def test_isomap_simple_grid():
-    # Isomap should preserve distances when all neighbors are used
-    N_per_side = 5
-    Npts = N_per_side ** 2
-    n_neighbors = Npts - 1
 
+def create_sample_data(add_noise=False):
     # grid of equidistant points in 2D, n_components = n_dim
     X = np.array(list(product(range(N_per_side), repeat=2)))
+    if add_noise:
+        # add noise in a third dimension
+        rng = np.random.RandomState(0)
+        noise = 0.1 * rng.randn(Npts, 1)
+        X = np.concatenate((X, noise), 1)
+    return X
+
+
+@pytest.mark.parametrize(
+    "n_neighbors, radius",
+    [(Npts-1, None),
+     (None, np.inf)]
+)
+def test_isomap_simple_grid(n_neighbors, radius):
+    # Isomap should preserve distances when all neighbors are used
+    X = create_sample_data(add_noise=False)
 
     # distances from each point to all others
-    G = neighbors.kneighbors_graph(X, n_neighbors,
-                                   mode='distance').toarray()
+    if n_neighbors is not None:
+        G = neighbors.kneighbors_graph(X, n_neighbors,
+                                       mode="distance").toarray()
+    else:
+        G = neighbors.radius_neighbors_graph(X, radius,
+                                             mode="distance").toarray()
 
     for eigen_solver in eigen_solvers:
         for path_method in path_methods:
-            clf = manifold.Isomap(n_neighbors=n_neighbors, n_components=2,
+            clf = manifold.Isomap(n_neighbors=n_neighbors,
+                                  radius=radius,
+                                  n_components=2,
                                   eigen_solver=eigen_solver,
                                   path_method=path_method)
             clf.fit(X)
 
-            G_iso = neighbors.kneighbors_graph(clf.embedding_,
-                                               n_neighbors,
-                                               mode='distance').toarray()
+            if n_neighbors is not None:
+                G_iso = neighbors.kneighbors_graph(clf.embedding_,
+                                                   n_neighbors,
+                                                   mode='distance')
+            else:
+                G_iso = neighbors.radius_neighbors_graph(clf.embedding_,
+                                                         radius,
+                                                         mode='distance')
+            G_iso = G_iso.toarray()
             assert_array_almost_equal(G, G_iso)
 
 
-def test_isomap_reconstruction_error():
+@pytest.mark.parametrize(
+    "n_neighbors, radius",
+    [(24, None),
+     (Npts-1, np.inf)]
```

Comment:
```suggestion
    [(Npts-1, None),
     (None, np.inf)]
```

Reply:
Sorry I missed that spot, thanks a lot for catching it!